### ISSUE TITLE
Internationalization of usage help

### ISSUE THREAD

**Issue by dag** [pcapriotti/optparse-applicative#9]
I don't need this critically, but thought I'd suggest it anyway. The "obvious" way to do it seems to me to be having messages be algebraic types, probably one user-provided and one library-provided. Something like how [errors are handled in reform](http://www.happstack.com/docs/crashcourse/Reform.html).

**Comment by HuwCampbell** [comment]
I agree, this is something I've recently added to a similar scala project and will get to soon.
